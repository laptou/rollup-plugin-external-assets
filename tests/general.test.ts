import nodeResolve from "@rollup/plugin-node-resolve";
import { outputSnapshotMacro } from "./macros";
import externalAssets from "../src/index";

describe("falsy values as filter patterns", () => {
	const falsy = [undefined, null, false, "", NaN, 0];

	for (const value of falsy) {
		const stringified = value !== value ? "NaN" : JSON.stringify(value);

		test("" + stringified, () => {
			expect(() => externalAssets(value as any)).toThrow(Error);
		});
	}
});

test("Multiple instances of the plugin can be used at the same time", () => {
	return outputSnapshotMacro({
		input: "tests/fixtures/src/index2.js",
		plugins: [
			externalAssets("tests/fixtures/assets/*"),
			nodeResolve({
				moduleDirectories: ["tests/fixtures/node_modules"],
			}),
			externalAssets(/@fontsource\/open-sans/),
		],
	});
});
