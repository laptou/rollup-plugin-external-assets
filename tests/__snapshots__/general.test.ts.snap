// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple instances of the plugin can be used at the same time 1`] = `
Array [
  Object {
    "code": "import png from \\"./assets/image-0fc60877.png\\";
import text from \\"./assets/text-6d7076f2.txt\\";
import \\"./assets/styles-fc0ceb37.css\\";
import font from \\"./assets/open-sans-all-300-italic-abf6ec0a.woff\\";

// This import should not be removed unless specified by the user or some other plugin.

var sayHi = () => {
	console.log(\\"hello\\");
};

var _default = () => {
	console.log(png);
	console.log(text);
	sayHi();
};

var index2 = {
	default: _default,
	font,
	non_string_literals: [1929, /^ab|b$/g, null, false],
};

export default index2;
",
    "fileName": "index2.js",
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "name": "index2",
  },
]
`;

exports[`Multiple instances of the plugin can be used at the same time 2`] = `
Array [
  Object {
    "fileName": "assets/image-0fc60877.png",
    "name": "image.png",
  },
  Object {
    "fileName": "assets/text-6d7076f2.txt",
    "name": "text.txt",
  },
  Object {
    "fileName": "assets/styles-fc0ceb37.css",
    "name": "styles.css",
  },
  Object {
    "fileName": "assets/open-sans-all-300-italic-abf6ec0a.woff",
    "name": "open-sans-all-300-italic.woff",
  },
]
`;
